lib/sensors/led_driver.hpp

```diff
@@ -0,0 +1,42 @@
+#pragma once
+
+#include "i2c_sensors.hpp"
+
+#include <cstdint>
+#include <memory>
+#include <optional>
+
+#include <core/logger.hpp>
+#include <io/i2c.hpp>
+
+static constexpr std::uint8_t kDefaultLedDriverAddress = 0x30;
+
+namespace hyped::sensors {
+class LedDriver {
+ public:
+  static std::optional<LedDriver> create(core::ILogger &logger, std::shared_ptr<io::II2c> i2c);
+  ~LedDriver();
+
+  core::Result initialise();
+  core::Result setColour(std::uint8_t channel,
+                         std::uint8_t red,
+                         std::uint8_t green,
+                         std::uint8_t blue);
+  core::Result setIntensity(std::uint8_t channel, std::uint8_t intensity);
+  core::Result reset();
+
+ private:
+  LedDriver(core::ILogger &logger, std::shared_ptr<io::II2c> i2c);
```

Comment:
Constructor should be public

Reply:
although I'm not sure it needs to be if it is only called from create

Reply:
i feel like it has had to be public even with the create, but i dont remembeer why. if it builds it builds